### DISCUSSION TITLE
More than 1 Slave modbus tcp

### DISCUSSION THREAD

**Discussion by Gnec220** [wlcrs/huawei_solar#529] [General]
Hi,

I connected my Huawei setup with HA via Modbus TCP. Now i need to connect EVCC( Solar Electric vehicles management) software with Huawei. I observed that Home assistant integration loses connection.

Thats why i would like to know if its possible to connect multiple slaves to the Huawei modbus TCP connection  which can obtain data parallely.

thank you in advance.

**Comment by Roving-Ronin** [comment]
@Gnec220 assume your using the USB dongle ? If so, no the dongle only allows a single modbus_tcp client.  Suggest you look at the modbus proxy https://github.com/Akulatraxas/ha-modbusproxy that you can add in HACS. With this you point the proxy to connect to the dongle, then point wlcrs and your EVCC to point to the Proxy.

Which EV charger are you using ? Assume your referring to this software: https://community.home-assistant.io/t/charge-your-ev-with-unused-solar-with-evcc/505033

**Reply by Gnec220** [reply]
Thank you. I used the modbus proxy solution in between. 

I have a pulsar plus from wallbox chargers. I initially used evcc mentioned from the link. But decided to use simply the wallbox app.
